pypesto/hierarchical/petab.py

```diff
@@ -504,3 +521,139 @@ def _get_symbolic_formula_from_measurement(
+    # and their corresponding observables in the measurement table.
+    relative_observables = set()
+    if PARAMETER_TYPE in petab_problem.parameter_df.columns:
+        par_df = petab_problem.parameter_df[
+            petab_problem.parameter_df[PARAMETER_TYPE].notna()
+        ]
+        meas_df_w_obs_pars = meas_df[meas_df[OBSERVABLE_PARAMETERS].notna()]
+        for par_id, row in par_df.iterrows():
+            if not row[ESTIMATE]:
+                continue
+            if row[PARAMETER_TYPE] in [
+                InnerParameterType.SCALING,
+                InnerParameterType.OFFSET,
+            ]:
+                for _, row in meas_df_w_obs_pars.iterrows():
+                    if par_id in row[OBSERVABLE_PARAMETERS].split(
+                        PARAMETER_SEPARATOR
+                    ):
+                        relative_observables.add(row[OBSERVABLE_ID])
+
+            elif row[PARAMETER_TYPE] == InnerParameterType.SIGMA:
+                corresponding_obs = set(
+                    meas_df[meas_df[NOISE_PARAMETERS] == par_id][OBSERVABLE_ID]
+                )
+                # If a sigma parameter belongs to a semi-quantiative
+                # observable, it is not a relative inner parameter.
+                if SEMIQUANTITATIVE in observables_by_data_type and (
+                    corresponding_obs
+                    & observables_by_data_type[SEMIQUANTITATIVE]
+                ):
+                    continue
+                relative_observables.update(corresponding_obs)
+
+    observables_by_data_type[RELATIVE] = relative_observables
+
+    # Check whether there is any overlap across data types
+    for data_type, observables in observables_by_data_type.items():
+        for (
+            other_data_type,
+            other_observables,
+        ) in observables_by_data_type.items():
+            if data_type == other_data_type:
+                continue
+            if observables & other_observables:
+                raise ValueError(
+                    "The same observable is associated with multiple data "
+                    f"types. Observable(s): `{observables & other_observables}`. "
+                    f"Data types: `{data_type}`, `{other_data_type}`."
+                )
+
+    # Validate ordinal measurement specification
+    if ORDINAL in observables_by_data_type:
+        meas_df_w_ordinals = meas_df[meas_df[MEASUREMENT_TYPE] == ORDINAL]
+        if MEASUREMENT_CATEGORY not in meas_df_w_ordinals.columns:
+            raise ValueError(
+                "Measurement category must be specified for ordinal "
+                "measurements."
+            )
+        for _, row in meas_df_w_ordinals.iterrows():
+            try:
+                int(row[MEASUREMENT_CATEGORY])
+            except ValueError as e:
+                raise ValueError(
+                    "Measurement category for ordinal measurement must be "
+                    "an integer."
```

Comment:
This will let 1.23 pass as integer.

Better check `float(...).is_integer()`

Reply:
Reorganize/Order import statements?

Reply:
> This will let 1.23 pass as integer.
> 
> Better check `float(...).is_integer()`

Think this solves it:
```
category = float(row[MEASUREMENT_CATEGORY])
if category != int(category):
    raise ValueError
```